runtime/centrifuge/src/governance/tracks.rs

```diff
@@ -0,0 +1,192 @@
+///
+/// NOTE: All curves are by definition normalized to
+/// * SUP(0) = 50%
+/// * SUP(2) = 46.08%
+/// * SUP(25) = 1%
+/// * SUP(28) = 1%
+const SUP_TREASURER: Curve = Curve::make_linear(25, 28, percent(1), percent(50));
+const APP_TREASURER: Curve = Curve::make_linear(14, 14, percent(70), percent(100));
+
+/// * SUP(0) = 50%
+/// * SUP(1/8) = 20%
+/// * SUP(1) = 3.85%
+/// * SUP(7) = 0.59%
+const SUP_REF_CANCELLER: Curve = Curve::make_reciprocal(1, 7, ppm(38500), ppm(5900), percent(50));
+const APP_REF_CANCELLER: Curve = Curve::make_linear(7, 7, percent(70), percent(100));
+
+/// * SUP(0) = 50%
+/// * SUP(1/8) = 20%
+/// * SUP(1) = 3.85%
+/// * SUP(7) = 0.59%
+const SUP_REF_KILLER: Curve = Curve::make_reciprocal(1, 7, ppm(38500), ppm(5900), percent(50));
+const APP_REF_KILLER: Curve = Curve::make_linear(7, 7, percent(70), percent(100));
+
+const TRACKS_DATA: [(u16, pallet_referenda::TrackInfo<Balance, BlockNumber>); 6] = [
+	(
+		0,
+		pallet_referenda::TrackInfo {
+			name: "root",
+			max_deciding: 1,
+			decision_deposit: 300_000 * CFG,
+			prepare_period: 6 * HOURS,
+			decision_period: 14 * DAYS,
+			confirm_period: 12 * HOURS,
+			min_enactment_period: 24 * HOURS,
+			min_approval: APP_ROOT,
+			min_support: SUP_ROOT,
+		},
+	),
+	(
+		1,
+		pallet_referenda::TrackInfo {
+			name: "whitelisted_caller",
+			max_deciding: 20,
+			decision_deposit: 1_000 * CFG,
+			prepare_period: 10 * MINUTES,
+			decision_period: 7 * DAYS,
+			confirm_period: 10 * MINUTES,
+			min_enactment_period: 10 * MINUTES,
+			min_approval: APP_WHITELISTED,
+			min_support: SUP_WHITELISTED,
+		},
+	),
+	(
+		2,
+		pallet_referenda::TrackInfo {
+			name: "pool_admin",
+			max_deciding: 5,
+			decision_deposit: 1_000 * CFG,
+			prepare_period: 1 * HOURS,
+			decision_period: 7 * DAYS,
+			confirm_period: 1 * HOURS,
+			min_enactment_period: 1 * HOURS,
+			min_approval: APP_POOL_ADMIN,
+			min_support: SUP_POOL_ADMIN,
+		},
+	),
+	(
+		3,
+		pallet_referenda::TrackInfo {
+			name: "treasurer",
+			max_deciding: 2,
+			decision_deposit: 10_000 * CFG,
+			prepare_period: 6 * HOURS,
+			decision_period: 14 * DAYS,
+			confirm_period: 12 * HOURS,
+			min_enactment_period: 12 * HOURS,
+			min_approval: APP_TREASURER,
+			min_support: SUP_TREASURER,
+		},
+	),
+	(
+		4,
+		pallet_referenda::TrackInfo {
+			name: "referendum_canceller",
+			max_deciding: 20,
+			decision_deposit: 50_000 * CFG,
+			prepare_period: 1 * HOURS,
+			decision_period: 7 * DAYS,
+			confirm_period: 1 * HOURS,
+			min_enactment_period: 10 * MINUTES,
+			min_approval: APP_REF_CANCELLER,
+			min_support: SUP_REF_CANCELLER,
+		},
+	),
+	(
+		5,
+		pallet_referenda::TrackInfo {
+			name: "referendum_killer",
+			max_deciding: 20,
+			decision_deposit: 75_000 * CFG,
+			prepare_period: 1 * HOURS,
+			decision_period: 7 * DAYS,
+			confirm_period: 1 * HOURS,
+			min_enactment_period: 10 * MINUTES,
+			min_approval: APP_REF_KILLER,
+			min_support: SUP_REF_KILLER,
+		},
+	),
+];
+
+pub struct TracksInfo;
```

Comment:
Could we abstract these files in `runtime-common`? Maybe we can pass some config type with the changed values in each runtime

Reply:
We could move three scopes into `runtime-common`:
1. `impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo` block
2. `fn percent`
3. `fn ppm`

The rest is completely custom to each runtime and I would imagine readability to suffer just for moving a few more lines into common.

Reply:
Makes sense!

Reply:
Unfortunately, we cannot move the large `impl pallet_referenda::TracksInfo<Balance, BlockNumber> for TracksInfo` block to common because it requires two inputs: `RuntimeOrigin` from the runtime and the `TRACKS_DATA` array which is custom to each runtime.

I tried to get around making both these parameters generic but the `pallet_referenda::impl_tracksinfo_get!(TracksInfo, Balance, BlockNumber)` macro then acts as a blocker 😞

Reply:
Then something like: 

```rust
pallet_referenda::impl_tracksinfo_get!(TracksInfo<Origin, TRACKS_DATA>, Balance, BlockNumber);
```

called in each runtime doesn't work, right?

No worries! Thanks for trying this! hopefully if we condense all runtimes into one in the future using a macro for the different configuration we can simplify this

Reply:
This is the best I could do: [a26245a](https://github.com/centrifuge/centrifuge-chain/pull/1828/commits/a26245a88cb9faadc3c0441092cffd4a89411859) + [eebb23f](https://github.com/centrifuge/centrifuge-chain/pull/1828/commits/eebb23f8a69b85043e0703a874f8900e4335ba53)